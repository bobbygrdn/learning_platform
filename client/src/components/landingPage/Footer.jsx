import React from 'react';
import '../../styles/landingPage/Footer.css';

export default function Footer() {
    return (
        <footer data-testid="footer">
            <p className="copyright" data-testid="copyright">Copyright Â© 2023 Valor Academy. All rights reserved.</p>
            <p className="media" data-testid="media">Follow us on social media for news, updates, and promotions.</p>
            <ul data-testid="ul">
                <li data-testid="li">
                    <a href="/terms" data-testid="anchor">Terms of Service</a>
                </li>
                <li data-testid="li">
                    <a href="/privacy" data-testid="anchor">Privacy Policy</a>

                </li>
                <li data-testid="li">
                    <a href="https://github.com/bobbygrdn/robert_gordon_capstone.git" data-testid="anchor">GitHub</a>
                </li>
                <li data-testid="li">
                    <a href="https://bobbygrdn.atlassian.net/wiki/spaces/VA/pages/7012354/Documentation?atlOrigin=eyJpIjoiODRkZmIzNWU2YjIxNGEyYjliZDAyMTVlYjExN2QyN2QiLCJwIjoiaiJ9" data-testid="anchor">Jira</a>
                </li>
            </ul>
        </footer>
    )
}
