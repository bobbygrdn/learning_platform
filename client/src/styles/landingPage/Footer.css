footer {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background-color: #677d8c;
  width: 100%;
  font-size: 28px;
  padding-bottom: 1em;
  text-align: center;
  color: var(--primary-color);
}

.copyright {
  margin-top: 1em;
}

ul {
  list-style-type: none;
  display: flex;
  flex-direction: row;
  margin-top: 0.5em;
}

li {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding-right: 0.5em;
  padding-left: 0.5em;
}

li a {
  color: var(--primary-color);
  text-decoration: none;
}

.break {
  margin-left: 0.5em;
  font-size: 28px;
}

@media screen and (max-width: 600px) {
  footer {
    font-size: 12px;
  }
}
