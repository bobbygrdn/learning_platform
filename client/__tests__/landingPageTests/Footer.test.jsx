import React from 'react';
import { render, screen } from '@testing-library/react';
import Footer from '../../src/components/landingPage/Footer';


it('renders without crashing', () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
})

it('should render paragraph tag with copyright classname', () => {
    render(<Footer />);
    const paragraphs = screen.getAllByTestId('copyright');
    expect(paragraphs).toHaveLength(1);
})

it('should render footer element', () => {
    render(<Footer />);
    const footer = screen.getAllByTestId('footer');
    expect(footer).toHaveLength(1);
})

it('should render paragraph tag with media classname', () => {
    render(<Footer />);
    const paragraphs = screen.getAllByTestId('media');
    expect(paragraphs).toHaveLength(1);
})

it('should render unorganized list tag', () => {
    render(<Footer />);
    const unorganizedList = screen.getAllByTestId('ul');
    expect(unorganizedList).toHaveLength(1);
})

it('should render 4 list item tags', () => {
    render(<Footer />);
    const listItems = screen.getAllByTestId('li');
    expect(listItems).toHaveLength(4);
})

it('should render 4 anchor tags', () => {
    render(<Footer />);
    const anchors = screen.getAllByTestId('anchor');
    expect(anchors).toHaveLength(4);
})