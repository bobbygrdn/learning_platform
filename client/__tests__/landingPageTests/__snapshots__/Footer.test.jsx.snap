// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <footer
    data-testid="footer"
  >
    <p
      class="copyright"
      data-testid="copyright"
    >
      Copyright Â© 2023 Valor Academy. All rights reserved.
    </p>
    <p
      class="media"
      data-testid="media"
    >
      Follow us on social media for news, updates, and promotions.
    </p>
    <ul
      data-testid="ul"
    >
      <li
        data-testid="li"
      >
        <a
          data-testid="anchor"
          href="/terms"
        >
          Terms of Service
        </a>
      </li>
      <li
        data-testid="li"
      >
        <a
          data-testid="anchor"
          href="/privacy"
        >
          Privacy Policy
        </a>
      </li>
      <li
        data-testid="li"
      >
        <a
          data-testid="anchor"
          href="https://github.com/bobbygrdn/robert_gordon_capstone.git"
        >
          GitHub
        </a>
      </li>
      <li
        data-testid="li"
      >
        <a
          data-testid="anchor"
          href="https://bobbygrdn.atlassian.net/wiki/spaces/VA/pages/7012354/Documentation?atlOrigin=eyJpIjoiODRkZmIzNWU2YjIxNGEyYjliZDAyMTVlYjExN2QyN2QiLCJwIjoiaiJ9"
        >
          Jira
        </a>
      </li>
    </ul>
  </footer>
</div>
`;
